libs/libc/pthread/pthread_exit.c

```diff
@@ -55,6 +55,10 @@
 
 void pthread_exit(FAR void *exit_value)
 {
+  FAR struct tls_info_s *info = tls_get_info();
+
+  info->flags |= TLS_THREAD_EXIT_PROCESSING;
```

Comment:
@xiaoxiang781216 if you were to use TCB_FLAG_EXIT_PROCESSING here, it would prevent
running nxtask_exithook from nx_pthread_exit, as the flag is tested in nxtask_exithook:
https://github.com/apache/nuttx/blob/089b5e07c7ebb80d5c60058eff2214f51e5895c8/sched/task/task_exithook.c#L414-L435

Reply:
Yes, I think TCB_FLAG_EXIT_PROCESSING could reuse to avoid pthread_exit more than once

Reply:
But the problem is the calling order:
pthread_exit() ->
nx_pthread_exit() ->
nxtask_exithook()

If TCB_FLAG_EXIT_PROCESSING is set in pthread_exit, nxtask_exithook will never run and a leak occurs.

Reply:
after this patch:
https://github.com/apache/nuttx/pull/13728/files#diff-bebcfde51101428c2031392f8f356093b1ca4bc86fea32f4aafe45ba9c0c3d12R432
it's not a problem anymroe.

Reply:
> after this patch: https://github.com/apache/nuttx/pull/13728/files#diff-bebcfde51101428c2031392f8f356093b1ca4bc86fea32f4aafe45ba9c0c3d12R432 it's not a problem anymroe.

This change is to prevent tls_cleanup from entering twice, the PR above does not solve this problem

Reply:
Yes, but nxsig_abnormal_termination can check TCB_FLAG_EXIT_PROCESSING instead adding a new flag.